From c0f7bc33ead3711d010e1057c236c5f500246449 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 5 May 2014 19:52:31 -0700
Subject: [PATCH 366/793] drm: xilinx: connector: Call the encoder slave's
 mode_valid() function

Call the mode valid function of the encoder slave, and let the encoder
slave driver check validity of given mode.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_connector.c |   15 ++++++++++-----
 1 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_connector.c b/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
index dfad0e1..273717d 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_connector.c
@@ -54,13 +54,18 @@ static int xilinx_drm_connector_get_modes(struct drm_connector *base_connector)
 static int xilinx_drm_connector_mode_valid(struct drm_connector *base_connector,
 					   struct drm_display_mode *mode)
 {
-	if (mode->clock > 165000)
-		return MODE_CLOCK_HIGH;
+	struct xilinx_drm_connector *connector =
+		to_xilinx_connector(base_connector);
+	struct drm_encoder *encoder = connector->encoder;
+	struct drm_encoder_slave *encoder_slave = to_encoder_slave(encoder);
+	struct drm_encoder_slave_funcs *encoder_sfuncs =
+		encoder_slave->slave_funcs;
+	int ret = MODE_OK;
 
-	if (mode->flags & DRM_MODE_FLAG_INTERLACE)
-		return MODE_NO_INTERLACE;
+	if (encoder_sfuncs->mode_valid)
+		ret = encoder_sfuncs->mode_valid(encoder, mode);
 
-	return MODE_OK;
+	return ret;
 }
 
 /* find best encoder: return stored encoder */
-- 
1.7.5.4

